### PR TITLE
refactor sentry section 

### PR DESCRIPTION
**Pull request by egabancho**
invenio: refactor sentry block

    * Deprecates non-camelcase sentry variables.

    * Addresses secret handling as described in #117.

### CODE REVIEW

**Comment by egabancho** [comment]
I know we should be doing topical PR, but I can't make all these changes in separate branches and then merge them without spending time resolving many conflicts.
I'll do a commit for each section within `invenio` Hopefully this helps with the review process.

File: charts/invenio/templates/NOTES.txt
Comment (lindhe) [inline charts/invenio/templates/NOTES.txt:13]:
```suggestion
{{- if .Values.invenio.sentry.existing_secret }}
```

File: charts/invenio/templates/NOTES.txt
Comment (lindhe) [inline charts/invenio/templates/NOTES.txt:22]:
```suggestion
{{- if .Values.invenio.sentry.secret_name }}
```

File: charts/invenio/templates/sentry-secret.yaml
Comment (lindhe) [inline charts/invenio/templates/sentry-secret.yaml:1]:
```suggestion
{{- if and (.Values.invenio.sentry.enabled) (and (not .Values.invenio.sentry.existingSecret) (not .Values.invenio.sentry.existing_secret)) }}
```

Comment (lindhe) [inline charts/invenio/templates/sentry-secret.yaml:1]:
```suggestion
{{- if and (.Values.invenio.sentry.enabled) (and (not .Values.invenio.sentry.existingSecret) (not .Values.invenio.sentry.existing_secret)) }}
```

File: charts/invenio/templates/sentry-secret.yaml
Comment (lindhe) [inline charts/invenio/templates/sentry-secret.yaml:2]:
```suggestion
{{- $dsn := .Values.invenio.sentry.dsn | required ".Values.invenio.sentry.dns is required unless secret provided." -}}
```

File: charts/invenio/values.yaml
Comment (lindhe) [inline charts/invenio/values.yaml:25]:
```suggestion
    ## @param invenio.sentry.dns Sentry DSN, required unless secret is provided
```

File: charts/invenio/values.yaml
Comment (lindhe) [inline charts/invenio/values.yaml:37]: > Only used when `invenio.sentry.existingSecret` is set.

Is that true?

File: charts/invenio/templates/sentry-secret.yaml
Comment (egabancho) [reply charts/invenio/templates/sentry-secret.yaml]: I wrote ".Values.invenio.sentry.dns is required unless existingSecret is provided." I find it more explicit. WDYT?

File: charts/invenio/values.yaml
Comment (egabancho) [reply charts/invenio/values.yaml]: Well ... no 😂 I believe it was the result of too much copy/paste from #117

File: charts/invenio/templates/NOTES.txt
Comment (lindhe) [reply charts/invenio/templates/NOTES.txt]: Unfortunately not. I think the Kubernetes extension in VSCode has support for syntax highlighting, not sure about linting. There are some other extensions too, but I have not used any of them.

I have explored many Vim plugins for Helm, they are all terrible. 😕

File: charts/invenio/templates/sentry-secret.yaml
Comment (lindhe) [reply charts/invenio/templates/sentry-secret.yaml]: Much better! I mostly corrected you because "if not secret is" sounded incorrect.  Your new version is great.

**Review by lindhe — APPROVED** [review]
Since I don't use Sentry, I cannot test these changes. But as far as I can tell, it looks good and as long as you test it before merging you have my approval. 👍

**Comment by egabancho** [comment]
I'm testing all the changes on our test instance. I will have to spin up a new one from scratch after the holiday break and test it there as well 😄

**Pull request merged**
